apps/web/src/pages/templates/components/VariantsPage.tsx

```diff
@@ -49,14 +65,136 @@ export function VariantsPage() {
     viewport.current.scrollTo({ top: scrollPosition.y > 0 ? 0 : viewport.current.scrollHeight, behavior: 'smooth' });
   };
 
+  const variants = step?.variants ?? [];
+
+  const { hasActiveIntegration, isChannelStep, activeIntegrationsByEnv } = useHasActiveIntegrations({
+    filterByEnv: true,
+    channelType: channel as unknown as ChannelTypeEnum,
+  });
+  const { primaryIntegration, isPrimaryStep } = useGetPrimaryIntegration({
+    filterByEnv: true,
+    channelType: channel as unknown as ChannelTypeEnum,
+  });
+
+  const missingProviderError = useMemo(() => {
+    if (!hasActiveIntegration && channel) {
+      return { errorMsg: `Provider is missing!`, errorType: NODE_ERROR_TYPES.MISSING_PROVIDER };
+    }
+
+    if (isPrimaryStep && !primaryIntegration) {
+      return { errorMsg: `Primary provider is missing!`, errorType: NODE_ERROR_TYPES.MISSING_PRIMARY_PROVIDER };
+    }
+
+    return null;
+  }, [hasActiveIntegration, channel, isPrimaryStep, primaryIntegration]);
+
+  const { variantsErrors, variantErrorsCount, variantsErrorMap } = useMemo(() => {
+    const rootVariantIndex = variants.length;
```

Comment:
Assigning an index to the Root step, this helps in the error navigation for the root step